tests/unit/sidecars/live_data_watcher_tests.py

```diff
@@ -42,7 +57,32 @@ def run(self, result: unittest.TestResult = None) -> unittest.TestResult:
             self.output_dir = Path(loc)
             return super().run(result)
 
-    def test_s3_load_type_on_change(self):
+    def test_generate_sharded_file_key_no_sharding(self):
+        original_file_key = "test_file_key"
+        actual_sharded_file_key = _generate_sharded_file_key(None, original_file_key)
```

Comment:
Do you want to extend this test to also include this condition?

```suggestion
        actual_sharded_file_key = _generate_sharded_file_key(1, original_file_key)
```

Reply:
And I guess also this?

```suggestion
        actual_sharded_file_key = _generate_sharded_file_key(0, original_file_key)
```

Since `not 0` should be truthy?

Reply:
nope, good catch

Reply:
wait, sorry I misunderstood.

Yeah it should be truthy - so if the value is 0, the prefix is ""